src/main/resources/category/java/common.xml

```diff
@@ -930,6 +930,192 @@ public RetrieveCache(final @Value("${cache.expiryTimeMillis}") long timeToLiveMi
+    or pmd-java:hasAnnotation('javax.ejb.Singleton') or pmd-java:hasAnnotation('jakarta.ejb.Singleton')
+    or pmd-java:hasAnnotation('org.springframework.stereotype.Component') or pmd-java:hasAnnotation('org.springframework.stereotype.Controller')
+    or pmd-java:hasAnnotation('org.springframework.stereotype.Service') or pmd-java:hasAnnotation('org.springframework.stereotype.Repository')
+    or pmd-java:hasAnnotation('org.springframework.context.annotation.Configuration')
+    or pmd-java:hasAnnotation('org.springframework.web.bind.annotation.RestController') or pmd-java:hasAnnotation('org.springframework.web.bind.annotation.ControllerAdvice')
+    or pmd-java:hasAnnotation('org.springframework.ws.server.endpoint.annotation.Endpoint')
+    or pmd-java:hasAnnotation('javax.persistence.Entity') or pmd-java:hasAnnotation('jakarta.persistence.Entity')
+    or pmd-java:hasAnnotation('javax.persistence.Embeddable') or pmd-java:hasAnnotation('jakarta.persistence.Embeddable')
+    or pmd-java:hasAnnotation('javax.persistence.MappedSuperclass') or pmd-java:hasAnnotation('jakarta.persistence.MappedSuperclass')
+    or ModifierList/Annotation[starts-with(@SimpleName, 'Xml')]
+)]
+[@Interface=false() and @Abstract=false() and not(pmd-java:typeIs('java.util.concurrent.Executor') or pmd-java:typeIs('java.lang.Throwable'))]
+/ClassBody[FieldDeclaration[@Static=false() and (not (pmd-java:hasAnnotation('javax.xml.bind.annotation.XmlElement')))
+    and
+    (: a getter with result type of a field :)
+    (ancestor::ClassBody[1]/MethodDeclaration
+       [(@Visibility='public' or ../../@Nested=true()) and @Static=false() and (starts-with(@Name, 'get') and string-length(@Name) > 3 or starts-with(@Name, 'is') and string-length(@Name) > 2)]
+       [ClassType/@SimpleName=ancestor::ClassBody[1]/FieldDeclaration[@Static=false()]/ClassType/@SimpleName]
+    (: or a @Getter on class :)
+    or ancestor::ClassDeclaration[1][pmd-java:hasAnnotation('lombok.Getter')]
+    )
+    (: and class has no equals nor hashCode method :)
+    and (not (ancestor::ClassBody[1]/MethodDeclaration[(@Visibility='public' or ../../@Nested=true()) and @Static=false() and (@Name='equals' and .//FormalParameters[@Size=1]) or (@Name='hashCode' and .//FormalParameters[@Size=0])]))
+    (: and class has a toString and #fields <= 1+#getters :)
+    and ((ancestor::ClassBody[1]//MethodDeclaration[(@Visibility='public' or ../../@Nested=true()) and @Static=false() and @Name='toString']
+            and count(ancestor::ClassBody[1]/FieldDeclaration[@Static=false()]) <=
+                (1 + count(ancestor::ClassBody[1]/MethodDeclaration[(@Visibility='public' or ../../@Nested=true()) and @Static=false() and (starts-with(@Name, 'get') and string-length(@Name) > 3 or starts-with(@Name, 'is') and string-length(@Name) > 2)]))
+            or ancestor::ClassDeclaration[1][ends-with(@SimpleName, 'Dto')])
```

Comment:
check with `to-upper` to also catch `fooDTO` classes?

Reply:
good point, added upper-case()